packages/DatePicker/src/DatePicker.js

```diff
@@ -140,14 +154,18 @@ function DatePicker(props) {
     window.requestAnimationFrame(() => {
       if (!isElementContainsFocus(calendarRef.current)) {
         handleInputConfirm();
-        hideCalendar();
       }
     });
   }
 
+  function handleInputChange(e) {
+    setInputtedString(e.target.value);
+  }
+
   function handleKeyUp(event) {
     if (event.key === "Enter") {
       handleInputConfirm();
+    } else if (event.key === "Escape") {
```

Comment:
This only works if the focus is on the Input, since the handler is attached to that component. If you add the handler to the Popover, I think it might work better for those users that are using the mouse, but want the esc shortcut to work anyway.